dataprofiler/profilers/column_profile_compilers.py

```diff
@@ -121,35 +121,43 @@ def update_profile(self, df_series):
         df_series = df_series.apply(str)
                 
         # If single process, loop and return
-        if not self.multiprocess_flag:
+        if pool is not None:
             for col_profile in self._profiles:
                 self._profiles[col_profile].update(df_series)
             return self
         
         # If multiprocess, setup pool, etc
         single_process_list = []
         multi_process_dict = {}
-        pool = mp.Pool(len(self._profilers))
         
         # Spin off seperate processes, where possible
         for col_profile in self._profiles:
-            try: # Add update function to be applied on the pool
-                multi_process_dict[col_profile] = pool.apply_async(
-                    self._profiles[col_profile].update, (df_series,))
-            except Exception as e: # Attempt again as a single process
+            single_thread_flag = (not self._profiles[col_profile].thread_safe)
+
+            if self._profiles[col_profile].thread_safe:
```

Comment:
I think this would be clearer as `if not single_thread_flag` just for readibility's sake. Not a big deal, just a suggestion